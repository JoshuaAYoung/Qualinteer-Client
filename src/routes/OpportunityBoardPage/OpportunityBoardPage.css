.oppBoardBanner {
  height: auto;
  margin: -8rem -2.5rem 0;
  padding: 8rem 2.5rem 0;
  background-color: #fff0b0;
  border-bottom: 0.2rem solid #f4bd5b;
}

.oppList {
  margin-bottom: 8rem;
}

.searchForm {
  position: relative;
  margin: 3.5rem 1rem 2rem;
}

.oppSearch[type='text'] {
  width: 100%;
  height: 3.5rem;
  font-size: 1.7rem;
  padding-left: 1rem;
  color: black;
  padding-bottom: 0.2em;
  background-color: white;
}

.searchButton {
  width: 4rem;
  height: 3.5rem;
  border-radius: 0;
  position: absolute;
  z-index: 1;
  right: 0;
  background-repeat: no-repeat;
  background-size: 2rem;
  background-position: center;
  background-image: url('/assets/mag-glass.svg');
}

.searchButton:hover {
  background-color: #f0667b;
}

.boardInstructions {
  text-align: center;
}

.opportunityCard {
  width: 100%;
  margin-top: 3.5rem;
}

.oppCardLink {
  background-color: #fff0b0;
  border-top: 0.2rem solid #f4bd5b;
  border-bottom: 0.2rem solid #f4bd5b;
  padding: 1rem;
  width: 100%;
  display: block;
  transition: all 0.5s;
}

.oppCardLink:hover {
  background-color: #f4bd5b;
}

.opportunityTitle {
  font-family: 'Montserrat', sans-serif;
  font-weight: 800;
  font-size: 1.7rem;
  margin: 0.2rem 0 0.7rem;
}

.oppCardLink {
  color: black;
  text-decoration: none;
  font-size: 1.4rem;
}

.orgInfo,
.opportunityDescription {
  margin-bottom: 2rem;
}

.nothingFound {
  font-size: 3rem;
  font-weight: 200;
  text-align: center;
  margin-top: 3rem;
  margin-bottom: 1rem;
}

.nothingLink {
  font-size: 1.8rem;
  font-weight: 300;
  color: black;
  text-align: center;
  display: block;
}

@media only screen and (min-width: 575px) {
  .oppList {
    padding: 0 8%;
  }

  .oppBoardBanner {
    padding-left: 15%;
    padding-right: 15%;
  }
}

@media only screen and (min-width: 768px) {
  .oppBoardBanner {
    margin-right: -4rem;
    margin-left: -4rem;
    padding-left: 25%;
    padding-right: 25%;
  }
}

@media only screen and (min-width: 1024px) {
  .oppBoardBanner {
    margin-right: calc(42rem - 49vw);
    margin-left: calc(42rem - 49vw);
    padding-left: calc(49vw - 27rem);
    padding-right: calc(49vw - 27rem);
  }

  .searchForm {
    margin-top: 5rem;
    margin-bottom: 3rem;
  }

  .boardInstructions {
    padding: 0 10rem;
  }

  .oppList {
    margin-top: 5rem;
    padding: 0;
  }
}
