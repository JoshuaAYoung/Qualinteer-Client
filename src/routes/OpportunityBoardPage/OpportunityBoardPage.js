import React, { Component } from 'react';
import OpportunityCard from '../../components/OpportunityCard/OpportunityCard'
import OppApiService from '../../services/opp-api-service'
import VableContext from '../../context/VableContext'
import './OpportunityBoardPage.css'
import ErrorBoundary from '../../ErrorBoundary'
import ErrorToast from '../../components/ErrorToast/ErrorToast'
import ScrollToTopOnMount from '../../utilities/ScrollToTopOnMount'

export default class OpportunityBoardPage extends Component {
  static contextType = VableContext;

  state = {
    error: null,
    tempSearchValue: ''
  }

  addTempSearch = (tempSearchValue) => {
    this.setState({
      tempSearchValue
    })
  }

  handleSubmit = ev => {
    ev.preventDefault()
    this.getOpportunities(this.state.tempSearchValue)
  }

  getOpportunities = (searchTerm) => {
    OppApiService.getOpps(searchTerm)
      .then(this.setState({ error: null }))
      .then(data => {
        if (data.length === 0) {
          this.setState({ error: 'No opportunities could be found.' })
        }
        this.context.setOppsBoard(data)
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  componentDidMount() {
    this.getOpportunities()
  }


  renderOpportunities() {
    return (
      <ul className='oppList'>
        {this.context.opportunities.map(opportunity =>
          <li key={opportunity.opportunity_id} className='opportunityCard'>
            <ErrorBoundary>
              <OpportunityCard
                opportunity={opportunity}
                history={this.props.history}
              />
            </ErrorBoundary>
          </li>
        )}
      </ul>
    )
  }

  render() {
    const { error } = this.state
    return (
      <div className='boardContainer'>
        <ScrollToTopOnMount />
        <div className="oppBoardBanner">
          <h1 className='pageHeader textCenter'>Opportunity Board.</h1>
          <p className='pageInstructions boardInstructions'>Click an opportunity to see more details or enter a search above.</p>
          <form className='searchForm' onSubmit={this.handleSubmit} aria-label='opportunity search'>
            <input
              placeholder='Keyword or Location...'
              type='text'
              name='oppSearch'
              id='oppSearch'
              className='oppSearch'
              aria-label="opportunity search"
              onChange={ev => this.addTempSearch(ev.target.value)}
            />
            <button
              type='submit'
              className='searchButton'
              aria-label='search button'>
            </button>
          </form>
        </div>
        {error && <ErrorToast errorMessage={error} />}
        {this.renderOpportunities()}
      </div >
    );
  }
}
