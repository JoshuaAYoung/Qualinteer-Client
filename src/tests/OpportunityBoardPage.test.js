import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import OpportunityBoardPage from '../routes/OpportunityBoardPage/OpportunityBoardPage';
import VableContext from '../context/VableContext'

describe('OpportunityBoardPage Component', () => {
  const opportunities = [
    {
      "opportunity_id": 1,
      "org_id": 3,
      "title": "Community Outreach Coordinator",
      "description": "This individual will work with the Executive Board Member in charge of Community Outreach to: Plan and implement outreach events Be RPI's connection to the community Cultivate relationships and collaborations with individuals, businesses, and other non-profit organizations doing similar work. Provide assistance to RPI's program assistants in supporting our constituents.", "posted": "2019-10-25T00:00:00.000Z",
      "usr_id": 1,
      "name": "Restoration Project International",
      "city": "Aurora",
      "state": "CO",
      "zipcode": 80010
    },
    {
      "opportunity_id": 2,
      "org_id": 3,
      "title": "Community Outreach Coordinator",
      "description": "This individual will work with the Executive Board Member in charge of Community Outreach to: Plan and implement outreach events Be RPI's connection to the community Cultivate relationships and collaborations with individuals, businesses, and other non-profit organizations doing similar work. Provide assistance to RPI's program assistants in supporting our constituents.", "posted": "2019-10-25T00:00:00.000Z",
      "usr_id": 2,
      "name": "Restoration Project International",
      "city": "Aurora",
      "state": "CO",
      "zipcode": 80010
    }
  ]

  it('OpportunityBoardPage Smoke Test', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <VableContext.Provider value={{ opportunities: opportunities }}>
        <OpportunityBoardPage />
      </VableContext.Provider>,
      div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('OpportunityBoardPage Snapshot Test', () => {
    const wrapper = shallow(
      <VableContext.Provider value={{ opportunities: opportunities }}>
        <OpportunityBoardPage />
      </VableContext.Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})