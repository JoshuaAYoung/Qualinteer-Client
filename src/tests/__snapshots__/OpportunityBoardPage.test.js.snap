// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityBoardPage Component OpportunityBoardPage Snapshot Test 1`] = `<OpportunityBoardPage />`;
